hls4ml/backends/catapult/catapult_backend.py

```diff
@@ -153,9 +153,8 @@ def _register_flows(self):
         ]
 
         if len(extras) > 0:
-            extras_flow = register_flow('extras', extras, requires=[init_flow], backend=self.name)
-        else:
-            extras_flow = None
+            for opt in extras:
+                print(f'WARNING: Optimizer "{opt}" is not part of any flow and will not be executed.')
```

Comment:
Is it better to print a warning or to do warnings.warn()? I am fine either way.

Reply:
There's a nice description about `warnings` vs `logging` in python documentation:
> [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning
>
> A logger’s [warning()](https://docs.python.org/3/library/logging.html#logging.Logger.warning) method if there is nothing the client application can do about the situation, but the event should still be noted

By this logic we should use `warnings.warn()` instead of our way of "logging" with `print()`. I'll update the PR.

On a related note, I was thinking that after we merge large PRs for 1.0 we do a simple pass and change all `print()` statements to `logging.info()` or `logging.warn()` and eventually use `logging.debug()` throughout the code to help with development/debugging. It was never a good time to do this as it affected large unmerged codebases, but it seems after 1.0 things will finally sufficiently quiet down for this to be done.